moped-editor/src/views/projects/projectView/ProjectComponentsMapEdit.js

```diff
@@ -338,17 +337,21 @@ const ProjectComponentsMapEdit = ({
     const removeAllSubcomponents = getAvailableSubcomponents().length === 0;
 
     // Generate a list of subcomponents to be removed
-    const removalList = subcomponentsDB.filter(
-      // Check every old subcomponent
-      oldSubcomponent =>
-        // If not found, mark it as true so it's part of the selectedSubcomponents list
-        removeAllSubcomponents ||
-        !selectedSubcomponents.find(
-          newSubcomponent =>
-            // Return true if we have found the old subcomponent in this list
-            oldSubcomponent.subcomponent_id === newSubcomponent.subcomponent_id
-        )
-    );
+    const removalList = subcomponentsDB
+      .filter(
+        // Check every old subcomponent
+        oldSubcomponent =>
+          // If not found, mark it as true so it's part of the selectedSubcomponents list
+          removeAllSubcomponents ||
+          !selectedSubcomponents.find(
+            newSubcomponent =>
+              // Return true if we have found the old subcomponent in this list
+              oldSubcomponent.subcomponent_id ===
+              newSubcomponent.subcomponent_id
+          )
+      )
+      // Mark remove list for deletion
+      .map(subcomponent => ({ ...subcomponent, is_deleted: true }));
```

Comment:
I refactored this code a bit to add the `is_deleted: true` k/v pair while generating the list of subcomponents that need removal. This way, we don't have to worry about using more logic below to search for where to add the flag since we know which ones need deletion right here.

Reply:
👍 much easier to grok!